### PR TITLE
Expose contract registry as a TS package

### LINKED ISSUE #161: Add typings to contract registry
- Add TypeScript typings to the npm package
- See [existing typings](https://github.com/nucypher/nucypher-ts/blob/main/packages/shared/src/contracts/registry.ts) for reference
- Consider generating these typings automatically form the registry JSON structure

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:** 
- 2

**What this does:**
- Adds TS configuration for a `@nucypher/nucypher-contracts` package
- Adds tests and utility methods originally developed in `@nucypher/shared` (`nucypher-ts`)
- Moves CI utils into `devDependencies` of `package.json` 

**Issues fixed/closed:**
- Fixes #161

**Why it's needed:**
- Moves responsibility for handling types away from `@nucypher/nucypher-contracts` users
- Adds utilities for testing & detecting compatibility changes

**Notes for reviewers:**
- Downstream changes: https://github.com/nucypher/nucypher-ts/pull/372

### CODE REVIEW

**Review by KPrasch — APPROVED** [review]
approving along with https://github.com/nucypher/nucypher-ts/pull/372/

**Comment by KPrasch** [comment]
See the published package here https://www.npmjs.com/package/@nucypher/nucypher-contracts?activeTab=code

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -40,23 +40,30 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
 
-      - name: Install Solhint and Prettier
-        run: npm install -g solhint solhint-plugin-prettier prettier
-
-      - name: Install Solidity plugin for Prettier
-        run: npm install prettier-plugin-solidity
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
```
Comment (cygnusv) [inline .github/workflows/main.yaml:56]: I see there's a very recent v4: https://github.com/actions/setup-node/releases/tag/v4.0.0

File: test/registry.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import { ChainId, type ContractName, type Domain, getContract } from '../src';
+
+const testCases: Array<[ string, number, ContractName ]> = [
+  [ 'lynx', 80001, 'Coordinator' ],
+  [ 'lynx', 80001, 'GlobalAllowList' ],
+  [ 'lynx', 80001, 'SubscriptionManager' ],
+  [ 'tapir', 80001, 'Coordinator' ],
+  [ 'tapir', 80001, 'GlobalAllowList' ],
+  [ 'tapir', 80001, 'SubscriptionManager' ],
```
Comment (cygnusv) [inline test/registry.spec.ts:11]: How about using `contractNames` from `src` instead of explicit contract names?

**Review by cygnusv — APPROVED** [review]

**Pull request merged**